collect_app/src/main/java/org/odk/collect/android/provider/ProjectsProvider.kt

```diff
@@ -0,0 +1,48 @@
+package org.odk.collect.android.provider
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.projects.ProjectsRepository
+import javax.inject.Inject
+
+class ProjectsProvider : ContentProvider() {
+
+    @Inject
+    lateinit var projectsRepository: ProjectsRepository
+
+    // Do not call it in onCreate() https://stackoverflow.com/questions/23521083/inject-database-in-a-contentprovider-with-dagger
+    private fun deferDaggerInit() {
```

Comment:
The naming of this threw me off a little as "defer" makes me think "do dagger init later" rather than "we are doing it later than expected". This could maybe be named "deferredDaggerInit"? Even simpler, it could just be `daggerInit` as I think your comment gives enough context.

Reply:
Fixed.